astropy/table/pprint.py

```diff
@@ -424,6 +424,7 @@ def _pformat_col_iter(
         """
         max_lines, _ = self._get_pprint_size(max_lines, -1)
         dtype = getattr(col, "dtype", None)
+        is_onedim = getattr(col, "ndim", 1) == 1
```

Comment:
I'd put this inside the `try/except` - having `ndim` and having a length are not the same.